### PR TITLE
RUBY-2363 Clear query cache on transaction commit and abort

### PR DESCRIPTION
**Pull request by egiurleo**

### CODE REVIEW

File: spec/integration/query_cache_spec.rb [hunk 1]
```diff
@@ -513,4 +513,113 @@
       expect(events.length).to eq(2)
     end
   end
+
+  describe 'in transactions' do
+    min_server_fcv '4.0'
+    require_topology :replica_set
```
Comment (p-mongo) [inline spec/integration/query_cache_spec.rb:519]: We have a `require_transaction_support` for this.

File: lib/mongo/session.rb [hunk 2]
```diff
@@ -602,6 +603,8 @@ def commit_transaction(options=nil)
     #
     # @since 2.6.0
     def abort_transaction
+      Mongo::QueryCache.clear_cache
```
Comment (p-mongo) [inline lib/mongo/session.rb:606]: `Mongo::` shouldn't be needed here.

**Review by p-mongo — COMMENTED** [review]
If one issues a query outside a transaction, starts a transaction and repeats the query, the query isn't repeated right? But if tx has snapshot read concern it would be repeated?

There should also be a docs update stating we clear cache on commit & abort, is that covered by another ticket?

File: spec/integration/query_cache_spec.rb (hunk 1 above)
Comment (egiurleo) [reply spec/integration/query_cache_spec.rb:519]: So convenient! Thanks for letting me know.

**Comment by egiurleo** [comment]
@p I added tests for both scenarios you mentioned and they work as you described. I also added a bullet point to [RUBY-2376](https://jira.mongodb.org/browse/RUBY-2376) about adding documentation for using the query cache in transactions.

**Review by p-mongo — APPROVED** [review]

File: spec/integration/query_cache_spec.rb [hunk 3]
```diff
@@ -513,4 +513,155 @@
...
+    require_transaction_support
+
+    let(:collection) { authorized_client['test'] }
+
+    let(:events) do
+      subscriber.command_started_events('find')
+    end
+
+    before do
+      collection.delete_many
```
Comment (p-mongo) [inline spec/integration/query_cache_spec.rb:527]: What do you think about making a helper like `create_collection` that would drop & create? Should take care of red CI then I expect.

**Pull request merged**